packages/ui/src/documents/EditDocument.tsx

```diff
@@ -96,17 +96,30 @@ export const EditDocument: FunctionComponent<Props> = (props: Props) => {
   }, [loadData]);
 
   const updateDocument = async (newDoc: Document) => {
+    let document;
     setState({
       loadingMessage: 'Updating document',
     });
     try {
-      const document = (await httpClient.documents.update(newDoc)).data;
+      /*
+      * We need to create a new version when updating a doc.
+      * TODO this might need to change if we do not auto commit anymore
+      * */
+      newDoc.document_id = newDoc!.header!.document_id
```

Comment:
Why do we need all these `!` hacks here? I think it would be good to either make these properties non-optional, or to add something like this above:

`if (!newDoc?.header) {
   throw new Error('Missing document fields')
}
`

Although the occasional `!` is fine, it would be better to avoid it whenever possible :)

Reply:
Wo we do not have any control on the typescript generated from the node swagger definitions. All props are optional there and typescript will complain. The best we can do is to do new types but that seems like an overhead and error prone. IMO this is better then having a lot of ifs to handle the optional definitions because we can control some of the props when we create them. We also have code were we add default values just to make sure those are not empty and we dot break the !